src/message/MQMessage.cpp

```diff
@@ -48,6 +48,9 @@ const string MQMessage::PROPERTY_TRANSACTION_CHECK_TIMES = "TRANSACTION_CHECK_TI
 const string MQMessage::PROPERTY_CHECK_IMMUNITY_TIME_IN_SECONDS = "CHECK_IMMUNITY_TIME_IN_SECONDS";
 
 const string MQMessage::KEY_SEPARATOR = " ";
+
+static const string EMPTY_STRING = "";
```

Comment:
I don't think it's the best approach to define empty str as a static global var.
Maybe we could add a new file `constant.h` like this:
```C++
class Constants {
public:
    static const string EMPTY_STRING = "";
};
```

Any way, we should avoid to use global variables and global static variables.

Reply:
@jovany-wang the visibility of global static variables is only in this file. i think depend on other file is unnecessary for only this one. if we use EMPTY_STRING in other files, it's ok.

Reply:
If you don't want to move it to `constant.h`, let's use `return "";` directly?
 Because I don't think we should use a global static variable in such case.:)

Reply:
> If you don't want to move it to `constant.h`, let's use `return "";` directly?
> Because I don't think we should use a global static variable in such case.:)

@jovany-wang now, we return a reference, and return a temp object is a bug.

Reply:
@ifplusor  Thanks, I see.
So if the method may be return a temp variable, it is pretty make sense to declare it as `string MQMessage::getProperty(const string& name);`.

If the method  always return his member variables, we could return a const ref.

Reply:
@jovany-wang i don't change method signature, we can discuss it.